xmodule/seq_block.py

```diff
@@ -557,10 +557,8 @@ def _get_render_metadata(self, context, children, prereq_met, prereq_meta_info,
                 'This section is a prerequisite. You must complete this section in order to unlock additional content.'
             )
 
-        blocks = self._render_student_view_for_blocks(context, children, fragment, view) if prereq_met else []
```

Comment:
@felipemontoya The `prereq_met` bit has been in place since [2018](https://github.com/openedx/edx-platform/commit/c4fc4b5df60c6eb6728e98f10c3dac57431cc3f6#diff-20826fcfc64f6ff8931e4b2e686a9a5a588b59c2eb808bb0eee0ed3ae4efa575L326) and was used to make sure the courseware didn't render blocks in gated content when the prerequisites aren't met.

Given the issue is new due to the migration to the MFE, are we sure we want to fix this by changing the behavior of the API? Shouldn't we fix this in the MFE?

Reply:
Hi @viadanna 

I understand. The problem is that I tried different ways from the MFE and I couldn't find a way to solve it because not having the unit of that section, which is a prerequisite, everything gets out of order at the route level because I don't have an identifier to reference in the route.

I understand that it is something that has been around since 2018 but since MFE is something new, perhaps the backend should have been adapted in that place so that it did not break MFE.

@dcoa @bra-i-am How do you see this?

Reply:
@jignaciopm Oh, I see, it's not an old bug, but the MFE requires an API behavior change. I'm worried that this would leak information to the client, as it would return the xblocks even when the content is gated. Can it gather just the blocks necessary for the MFE to work instead?